### PR TITLE
Add pull request workflow build

### PR DESCRIPTION
**Pull request by chathushkaayash**
This workflow automates building and verifying pull requests for the dev and main branches by setting up Java, Ballerina, and running a Gradle build.

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Pull request build
+
+on:
+  pull_request:
+    branches:
+      - dev
```
Comment (ayeshLK) [inline .github/workflows/pull-request.yml]: I think we only need to build the PRs sent for the `main` branch as that is the main git branch for a repository.

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+      # Set up JDK
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+
+      # Install Ballerina
+      - name: Install Ballerina
```
Comment (ayeshLK) [inline .github/workflows/pull-request.yml:41]: We have already specified the required Ballerina version in the `gradle.properties` file. Can you check how we can get that version here rather than hard-coding the version ?

File: ballerina/build.gradle [hunk 3]
```diff
@@ -1,6 +1,5 @@
-
 plugins {
-    id 'io.ballerina.plugin'
+    id "io.ballerina.plugin" version "${ballerinaGradlePluginVersion}"
```
Comment (ayeshLK) [inline ballerina/build.gradle:2]: Is this change related to workflows ? Best practice is to have one PR for a specific change.

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (chathushkaayash) [reply .github/workflows/pull-request.yml]: Ok I will remove the dev branch

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (chathushkaayash) [reply .github/workflows/pull-request.yml:41]:
```
# Read Ballerina version from gradle.properties
- name: Read Ballerina Version
  id: ballerina_version
  run: echo "ballerina_version=$(grep ballerinaVersion gradle.properties | cut -d'=' -f2)" >> $GITHUB_ENV

# Install Ballerina
- name: Install Ballerina
  uses: ballerina-platform/setup-ballerina@v1
  with:
    version: ${{ env.ballerina_version }}
    ```
          
I think this should work.

File: ballerina/build.gradle (hunk 3 above)
Comment (chathushkaayash) [reply ballerina/build.gradle:2]: What I should do when I working on one thing and I found out another small mistake that is not related to issue that I am working on, shouldn't I fix that small mistake? What should I do?

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (ayeshLK) [reply .github/workflows/pull-request.yml:41]: +1

File: ballerina/build.gradle (hunk 3 above)
Comment (ayeshLK) [reply ballerina/build.gradle:2]: This kind of things happen. But its best if we could send another PR for the fix (using another branch) and merge it to master before completing the feature you are working on.
